File: src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * 
 */
public class Monster extends LabyrinthCharacter {
    private static final int INITIAL_HEALTH=5;

    
    public Monster(String nombre, float inteligencia, float fuerza){
        super(nombre, fuerza, inteligencia, INITIAL_HEALTH);
    }
    @Override
    public float attack(){
        return Dice.intensity(this.getStrength());
    }
    @Override
    public boolean defend(float receivedAttack){
        boolean isDead = this.dead();
        if(!isDead){
            if(Dice.intensity(this.getIntellignece()) < receivedAttack){
                this.gotWounded();
                isDead = this.dead();
            }
        }
        return isDead;
    }
}

File: JUEGO-RUBY/irrgarten/Monster.rb
require_relative 'Dice'
require_relative 'LabyrinthCharacter'
module Irrgarten
    class Monster < LabyrinthCharacter
        @@INITIAL_HEALTH = 5
        def initialize(name, intelligence , strenght)
            super(name,intelligence,strenght,@@INITIAL_HEALTH)
        end
        
        def attack
            return Dice.intensity(@strenght)
        end
        def defend(recieved_attack)
            is_dead = self.dead
            if(!is_dead)
                defensive_energy = Dice.intensity(@intelligence)
                if(defensive_energy < recieved_attack)
                    self.got_wounded
                    is_dead = self.dead
                end
            end
            return is_dead

        end

    end
end
